---
title: "Data description"
format:
  html:
    embed-resources: true
editor: visual
---

# Load libraries

```{r}
#| warning: false
#| message: false

library("tidyverse")
library("cowplot")
library("broom")
library("table1")
library("ggplot2")
```

# Load data

```{r}
#| warning: false
#| message: false

augmented_table <- read_tsv("../data/03_augmented_table.tsv")
augmented_metadata <- read_tsv("../data/03_augmented_metadata.tsv")
PCA_ready_table <- read_tsv("../data/03_augmented_table_PCA.tsv")
```

# Table1

Overview comparison table of samples with glioma intra-tumoral tertiary lymphoid structures (TLSs) and control group samples.

```{r}
augmented_metadata_table2 <- augmented_metadata |>
  mutate(Gender = factor(Gender),
         Cohort = factor(Batch)) |>
  table1(x = formula(~ IDH_status + Gender + Age + Localisation | TLS_status),
         data = _)

htmltools::save_html(augmented_metadata_table2, 
                     file = "../results/TLS_status_table.html")

augmented_metadata_table2
```

Table mutational status of isocitrate dehydrogenase (IDH) against IDH wild type.

```{r}
augmented_metadata_table <- augmented_metadata |>
  mutate(Gender = factor(Gender),
         Cohort = factor(Batch)) |>
  table1(x = formula(~ TLS_status + Gender + Age + Localisation | IDH_status),
         data = _)

htmltools::save_html(augmented_metadata_table, 
                     file = "../results/IDH_status_table.html")

augmented_metadata_table
```

# PCA

Principle component analysis (PCA) of the augmented data was performed.

```{r}
pca_fit <- PCA_ready_table |> 
  select(where(is.numeric)) |> 
  select(!c(GEP_score, Age, Batch)) |>
  prcomp(scale = TRUE)

pca_for_plots <- pca_fit |> 
  augment(PCA_ready_table)
```

The first two PCs of the PCA conducted on the RNA-seq gene expression matrix stratified by TLS Status.

```{r}
pca_tls <- pca_for_plots |>
  ggplot(aes(.fittedPC1, 
             .fittedPC2, 
             color = TLS_status, 
             fill = TLS_status)) + 
  geom_point(size = 1.8, 
             alpha=1, 
             shape = 21) +
  scale_fill_manual(
    values = c(
      TLS    = "indianred3",
      control = "lightsteelblue2"
    )
  ) +
  scale_color_manual(
    values = c(
      TLS    = "indianred4",
      control = "lightsteelblue3"
    )
  ) +
  theme_half_open(12) +
  background_grid() +
  labs(
    color = "TLS status",
    fill  = "TLS status",
    title = "The first two PCs of transcription data, startified by TLS status",
    x = "PC1",
    y = "PC2"
  ) +
  theme(plot.title.position = "plot")

ggsave(
  filename = "../results/TLS_status_PCA.png",
  plot = pca_tls,
  device = "png",
  bg = "white"
)

pca_tls
```

The majority of the data points lay in the same broad cluster, both for control and TLS positive samples. There are some outliers for TLS positive samples that have a lower PC 2.

```{r}
pca_idh <- pca_for_plots |>
  ggplot(aes(.fittedPC1, .fittedPC2, color = IDH_status, fill = IDH_status)) +
  geom_point(size = 1.8,
             alpha = 1,
             shape = 21) +
  scale_fill_manual(
    values = c(
      IDH_mutant    = "indianred3",
      IDH_wild_type = "lightsteelblue2"
    ),
    labels = c(IDH_mutant = "IDH mutant", IDH_wild_type = "IDH wild type")
  ) +
  scale_color_manual(
    values = c(
      IDH_mutant    = "indianred4",
      IDH_wild_type = "lightsteelblue3"
    ),
    labels = c(IDH_mutant = "IDH mutant", IDH_wild_type = "IDH wild type")
  ) +
  theme_half_open(12) + 
  background_grid() +
  labs(
    color = "IDH status",
    fill  = "IDH status",
    title = "The first two PCs of transcription data, startified by IDH status",
    x = "PC1",
    y = "PC2"
  ) +
  theme(plot.title.position = "plot")

pca_idh
ggsave(
  filename = "../results/IDH_status_PCA.png",
  plot = pca_idh,
  device = "png",
  bg = "white"
)
```

In a PCA plot visualising the first two PCs of the RNA-seq expression data, stratified by IDH status, it can be clearly seen that the data points from IDH mutant samples and IDH wild type each cluster together. This is expected since the mutational status of the isocitrate dehydrogenase (IDH) induces broad metabolic and epigenetic reprogramming and therefore influences expression levels of other genes. It should be noted that the cumulative variance explained by PC1 & PC2 is about 20%.

```{r}
summary(pca_fit)$importance[3, 2]
```

```{r}
summary(pca_fit)$importance
```

```{r}
loadings <- pca_fit$rotation
```

Most important genes in proportion in the PC1 and PC2:

```{r}
top_PC1_pos <- loadings[, "PC1"] |> 
  sort(decreasing = TRUE) |> 
  head(20)
top_PC1_neg <- loadings[, "PC1"] |> 
  sort(decreasing = FALSE) |> 
  head(20)
```

```{r}
top_PC2_pos <- loadings[, "PC2"] |> 
  sort(decreasing = TRUE) |> 
  head(20)
top_PC2_neg <- loadings[, "PC2"] |> 
  sort(decreasing = FALSE) |> 
  head(20)
```

Variance proportion explained by all PCs plotted:

```{r}
var_explained <- summary(pca_fit)$importance["Proportion of Variance", ]
```

```{r}
df_var <- tibble(
  PC = seq_along(var_explained),
  variance = var_explained
)
```

```{r}
ggplot(df_var, 
       aes(x = PC, 
           y = variance)) +
  geom_point() +
  geom_line() +
  labs(
    title = "Variance explained by each principal component",
    x = "Principal Component",
    y = "Proportion of Variance"
  ) +
  theme_minimal()
```
